C#.md

```diff
@@ -0,0 +1,497 @@
+```c#
+private void DoSomething()
+{
+	// Bad.
+	long pollInterval;
+	int fileSize;
+}
+
+private void DoSomething()
+{
+	// Good.
+	long pollIntervalMs;
+	int fileSizeGb.
+}
+```
+
+**Constants**
+
+* Should be all uppercase with words separated by underscores (_).
+
+Examples:
+```c#
+// Bad.
+public const int TimeInSeconds = 5;
+
+//Good.
+public const int TIME_IN_SECONDS = 5;
+```
+
+**Private variables**
+
+* Should be written in **PascalCase**.
+* Should be prefixed with the **m_**
+
+Examples:
+```c#
+// Bad.
+int fileSize;
+
+// Good.
+int m_FileSizeGb.
+```
+
+**Local variables**
+
+* Should be written in **lowerCamelCase**.
+* Using the implicit typing for the local variables declaration is encouraged.
+
+Examples:
+```c#
+private void DoSomething()
+{
+	// Bad.
+	long PollInterval = Convert.ToInt32(Console.ReadLine());;
+}
+
+private void DoSomething()
+{
+	// Good.
+	var pollIntervalMs = Convert.ToInt32(Console.ReadLine());
+}
+```
+
+## Comments and Documentation
+
+Have you ever heard that "good code is supposed to be self-explanatory"? I'd love to find the author of this statement and tell him everything I think about it. This guy is responsible for thousands of unmaintainable projects because devs are lazy by nature and use it as excuse whenever it's possible.
+
+The problem is rather obvious: self-explanatory code only tell how it is working. It rarely tells how it should work. That's why we have some strict rules regarding code documentation and comments.
+
+The more visible a piece of code is (and by extension - the farther away consumers might be), the more documentation is needed.
+
+### General Documentation Quality
+
+* Use [XMLDoc](http://msdn.microsoft.com/en-us/library/b2s063f7%28v=VS.100%29.aspx)-style comments.
+* Every public class or method **must** have a comment.
+* Don't document overriding methods unless you want to tell what's different in its behavior.
+* Documentation text should be written using complete sentences, full stops are not necessary.
+
+### Documenting a Class
+Documentation for a class may range from a single sentence to paragraphs with code examples. Documentation should serve to disambiguate any conceptual blanks in the API, and make it easier to quickly and **correctly** use your API. A thorough class doc usually has a one sentence summary and, if necessary, a more detailed explanation.
+
+```C#
+/// <summary>
+/// Implements a variable-size List that uses an array of objects to store the
+/// elements. A List has a capacity, which is the allocated length
+/// of the internal array. As elements are added to a List, the capacity
+/// of the List is automatically increased as required by reallocating the
+/// internal array.
+/// </summary>
+public class List<T> : IList<T>, System.Collections.IList, IReadOnlyList<T>
+{
+	...
+}
+```
+
+### Documenting a Method
+
+A method doc should tell what the method does. Depending on the argument types, it may also be important to document input format.

```

Comment:
One more thing to add.

A good method comment should specify what exceptions can be thrown by the method. This requires quite a bit of work - you will need to go down to subsequent methods, but this will help a lot afterward.